### ISSUE TITLE
"Expect Error" comment

### ISSUE THREAD

**Issue by epsy** [python/mypy#13291]
**Feature**

A comment can be added to lines where the developer expects the type checker to raise an issue.
When typechecking:
- If the typechecker would have issued an error, no error is emitted
- If the typechecker wouldn't have issued an error, an error is emitted

```python
print(1 + "a")
print(1 + "a") # type: error
print(1 + 2) # type: error
```

mypy would report an error for lines 1 and 3.  (Of course, in this contrived example, running the code would throw on line 1)

**Pitch**

This is useful for:
- Cleaning up silenced expected errors as you resolve their root cause
  - Currently, using `# type: ignore` could lead you to keep those annotations beyond the moment they are no longer needed, making for a messy codebase.
  - For large codebases, migration between typechecker versions or settings can cause a lot of errors to show up.  It can be unrealistic to fix them all at once.  Adding `type: ignore` presents the same issue as above.
- Testing packages that define type annotations
  - They may need to test error cases in order to validate that their types are taken into account and strict enough.
  - They may need to show error examples and automatically validate that these examples remain up-to-date.

Prior work:
- [TypeScript also has these](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments).  We typically use them in migrations as described above, with a comment showing an excerpt of the error encountered and something like "FIXME" (TypeScript allows arbitrary text after `// @ts-expect-error`.)

**Comment by AlexWaygood** [comment]
You can already do this through a combination of mypy's [`--warn-unused-ignores`](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-warn-unused-ignores) setting and `type: ignore` comments. See some documentation [here](https://typing.readthedocs.io/en/latest/source/quality.html#testing-using-assert-type-and-warn-unused-ignores), and examples of tests that do just that [here](https://github.com/python/typeshed/blob/fdeb216547371d4301b5c001e31107e0ab1d0e23/test_cases/stdlib/builtins/test_pow.py#L78-L84)
